### PR TITLE
Remove references to fluentd logs configuration in otelcol logs configuration

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
- tests(fluentd): add tests for setting logs metadata
- chore!: move parameters from fluentd.logs.containers into sumologic.log.container
    - move `fluentd.logs.containers.sourceHost` to `sumologic.log.container.sourceHost`
    - move `fluentd.logs.containers.sourceName` to `sumologic.log.container.sourceName`
    - move `fluentd.logs.contianers.sourceCategory` to `sumologic.log.container.sourceCategory`
    - move `fluentd.logs.containers.sourceCategoryPrefix` to `sumologic.log.container.sourceCategoryPrefix`
    - move `fluentd.logs.contianers.sourceCategoryReplaceDash` to `sumologic.log.container.sourceCategoryReplaceDash`
    - move `fluentd.logs.containers.excludeContainerRegex` to `sumologic.log.container.excludeContainerRegex`
    - move `fluentd.logs.containers.excludeHostRegex` to `sumologic.log.container.excludeHostRegex`
    - move `fluentd.logs.containers.excludeNamespaceRegex` to `sumologic.log.container.excludeNamespaceRegex`
    - move `fluentd.logs.containers.excludePodRegex` to `sumologic.log.container.excludePodRegex`
    - move `fluentd.logs.containers.sourceHost` to `sumologic.log.container.sourceHost`
    - move `fluentd.logs.containers.perContainerAnnotationsEnabled` to `sumologic.log.container.perContainerAnnotationsEnabled`
    - move `fluentd.logs.containers.perContainerAnnotationPrefixes` to `sumologic.log.container.perContainerAnnotationPrefixes`

-  chore!: move parameters from fluentd.logs.kubelet into sumologic.logs.kubelet
    - move `fluentd.logs.kubelet.sourceName` to `sumologic.logs.kubelet.sourceName`
    - move `fluentd.logs.kubelet.sourceCategory` to `sumologic.logs.kubelet.sourceCategory`
    - move `fluentd.logs.kubelet.sourceCategoryPrefix` to `sumologic.logs.kubelet.sourceCategoryPrefix`
    - move `fluentd.logs.kubelet.sourceCategoryReplaceDash` to `sumologic.logs.kubelet.sourceCategoryReplaceDash`
    - move `fluentd.logs.kubelet.excludeFacilityRegex` to `sumologic.logs.kubelet.excludeFacilityRegex`
    - move `fluentd.logs.kubelet.excludeHostRegex` to `sumologic.logs.kubelet.excludeHostRegex`
    - move `fluentd.logs.kubelet.excludePriorityRegex` to `sumologic.logs.kubelet.excludePriorityRegex`
    - move `fluentd.logs.kubelet.excludeUnitRegex` to `sumologic.logs.kubelet.excludeUnitRegex`
  
-  chore!: move parameters from fluentd.logs.systemd to sumologic.log.systemd
    - move `fluentd.logs.systemd.sourceName` to `sumologic.logs.systemd.sourceName`
    - move `fluentd.logs.systemd.sourceCategory` to `sumologic.logs.systemd.sourceCategory`
    - move `fluentd.logs.systemd.sourceCategoryPrefix` to `sumologic.logs.systemd.sourceCategoryPrefix`
    - move `fluentd.logs.systemd.sourceCategoryReplaceDash` to `sumologic.logs.systemd.sourceCategoryReplaceDash`
    - move `fluentd.logs.systemd.sourceCategoryReplaceDash` to `sumologic.logs.systemd.sourceCategoryReplaceDash`
    - move `fluentd.logs.systemd.excludeFacilityRegex` to `sumologic.logs.systemd.excludeFacilityRegex`
    - move `fluentd.logs.systemd.excludeHostRegex` to `sumologic.logs.systemd.excludeHostRegex`
    - move `fluentd.logs.systemd.excludePriorityRegex` to `sumologic.logs.systemd.excludePriorityRegex`
    - move `fluentd.logs.systemd.excludeUnitRegex` to `sumologic.logs.systemd.excludeUnitRegex`
    
- chore!: move parameters from fluentd.logs.default to sumologic.logs.defaultFluentd
    - move `fluentd.logs.default.sourceName` to `sumologic.logs.defaultFluentd.sourceName`
    - move `fluentd.logs.default.sourceCategory` to `sumologic.logs.defaultFluentd.sourceCategory`
    - move `fluentd.logs.default.sourceCategoryPrefix` to `sumologic.logs.defaultFluentd.sourceCategoryPrefix`
    - move `fluentd.logs.default.sourceCategoryReplaceDash` to `sumologic.logs.defaultFluentd.sourceCategoryReplaceDash`
    - move `fluentd.logs.default.excludeFacilityRegex` to `sumologic.logs.defaultFluentd.excludeFacilityRegex`
    - move `fluentd.logs.default.excludeHostRegex` to `sumologic.logs.defaultFluentd.excludeHostRegex`
    - move `fluentd.logs.default.excludePriorityRegex` to `sumologic.logs.defaultFluentd.excludePriorityRegex`
    - move `fluentd.logs.default.excludeUnitRegex` to `sumologic.logs.defaultFluentd.excludeUnitRegex`

### CODE REVIEW

**Review by swiatekm-sumo — APPROVED** [review]
This all looks good to me, but I'd prefer if someone else reviewed it as well, since it's a pretty big change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,46 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - feat(logs)!: simplify metadata configuration [#2626]
 - feat(metrics)!: simplify metadata configuration [#2622]
 - feat(events)!: add config.merge option [#2643]
+- chore!: move parameters from `fluentd.logs.containers` to `sumologic.log.container` [#2635]
```
Comment (rnishtala-sumo) [inline CHANGELOG.md:31]: **nit:** should this be `sumologic.logs.container` instead of `sumologic.log.container`? same for the rest

Comment (kkujawa-sumo) [reply CHANGELOG.md]: Thank you! this is now corrected :)

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -252,12 +252,110 @@ sumologic:
...
+
+      ## A regular expression for facility.
+      ## Matching facility will be excluded from Sumo. The logs will still be sent to FluentD.
+      excludeFacilityRegex: ""
+      ## A regular expression for hosts.
+      ## Matching hosts will be excluded from Sumo. The logs will still be sent to FluentD.
+      excludeHostRegex: ""
+      ## A regular expression for priority.
+      ## Matching priority will be excluded from Sumo. The logs will still be sent to FluentD.
+      excludePriorityRegex: ""
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:354]: Why do we have `excludePriorityRegex` here and above?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:354]: Because fluentd uses it in `default` pipeline as well

**Review by sumo-drosiek — APPROVED** [review]
LGTM

File: deploy/helm/sumologic/conf/logs/fluentd/logs.source.default.conf [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
   @type kubernetes_sumologic
-  source_name {{ .Values.fluentd.logs.default.sourceName | quote }}
-  source_category {{ .Values.fluentd.logs.default.sourceCategory | quote }}
-  source_category_prefix {{ .Values.fluentd.logs.default.sourceCategoryPrefix | quote }}
-  source_category_replace_dash {{ .Values.fluentd.logs.default.sourceCategoryReplaceDash | quote }}
-  exclude_facility_regex {{ .Values.fluentd.logs.default.excludeFacilityRegex | quote }}
-  exclude_host_regex {{ .Values.fluentd.logs.default.excludeHostRegex | quote }}
-  exclude_priority_regex {{ .Values.fluentd.logs.default.excludePriorityRegex | quote }}
-  exclude_unit_regex {{ .Values.fluentd.logs.default.excludeUnitRegex | quote }}
+  source_name {{ .Values.sumologic.logs.defaultFluentd.sourceName | quote }}
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/conf/logs/fluentd/logs.source.default.conf:17]: Not sure if I understand this change - couldn't we just leave the original properties?

**Review by astencel-sumo — APPROVED** [review]
We need a migration for this :+1:

File: deploy/helm/sumologic/conf/logs/fluentd/logs.source.default.conf (hunk 3 above)
Comment (astencel-sumo) [reply deploy/helm/sumologic/conf/logs/fluentd/logs.source.default.conf:17]: The upside of this approach is consistency - everything is mostly in one place, under `sumologic.logs.*`.

**Pull request merged**